### PR TITLE
[AIRFLOW-3160] Load latest_dagruns asynchronously, speed up front page load time

### PR DESCRIPTION
**Pull request by aoen**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-3160\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3160
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-3160\], code changes always need a Jira issue.

### Description

The front page loads very slowly when the DB has latency because one blocking query is made per DAG against the DB. Even if we were to add indexes, the queries should still be batched to avoid an overhead of RTT * # of DAGs.

The latest dagruns should be loaded asynchronously and in batch like the other UI elements that query the database.

From my tests I was able to get the front page to reduce page load time of the front page from 8s to ~0.7s after this change with a DB in the west part of the USA with the Airflow cluster in the east part of the USA.

The load on the DB should also be slightly reduced due to this change as well.

- [X] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
UI changes only, tested on local webservers and our prod hosts.

### Commits
- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation
- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [X] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

cc @ashb

### CODE REVIEW

File: airflow/www_rbac/views.py [hunk 1]
```diff
@@ -392,6 +392,33 @@ def task_stats(self, session=None):
                     payload[dag.safe_dag_id].append(d)
         return wwwutils.json_response(payload)
 
+    @expose('/last_dagruns')
+    @has_access
+    @provide_session
+    def last_dagruns(self, session=None):
```
Comment (ashb) [inline airflow/www_rbac/views.py:398]: RBAC has different templates so nothing is calling this yet.

**Comment by ashb** [comment]
> The load on the DB should also be slightly reduced due to this change as well.

Aren't we still making all the same queries?

What's the time to the page being fully populated too please?

**Comment by aoen** [comment]
Good call on the RBAC templates, will make a fix. 

> Aren't we still making all the same queries?

Queries are slightly different, in the past we made one query per DAG, now it's just a single query, so the constant overhead of each query is removed.

> What's the time to the page being fully populated too please?

Locally the time for the page to be fully populated is pretty much instant. Prod is within a second, the query is pretty much a more lightweight version of the one being used to fetch and display task instance state on the same page so performance shouldn't be a concern.

**Comment by aoen** [comment]
@ashb addressed your comments, retrying CI now, but the two build jobs that failed out of nine were caused by CI flakiness.

File: airflow/www_rbac/views.py (hunk 1 above)
Comment (aoen) [reply airflow/www_rbac/views.py:398]: Fixed

**Comment by aoen** [comment]
CI has passed, ready to be merged.

**Comment by aoen** [comment]
Ready to merge FYI.

**Comment by aoen** [comment]
I'm happy to merge myself and take responsibility in fixing it if there is a breakage FYI.

@ashb

**Comment by Fokko** [comment]
Merged @aoen. High risk, high reward! 💪

**Comment by aoen** [comment]
@Fokko thanks! I owe you a PR review :).

**Comment by ashb** [comment]
@aoen Speak of the devil :D this seems to be causing test failures after this PR was merged, but only on MySQL somewhat odly. Can you take a look please, (we might revert this PR temporarily)

**Comment by ashb** [comment]
Examples that (we think) started happening after this PR was merged:

https://travis-ci.org/apache/incubator-airflow/jobs/451428747#L4660

in `ERROR: test_success (tests.www_rbac.test_views.TestAirflowBaseViews)` (postgres this time)

And same build on Mysql https://travis-ci.org/apache/incubator-airflow/jobs/451428748#L4660

Going to try reverting this PR and see if it fixes things, even though the error doesn't make any sense.

**Comment by aoen** [comment]
@ashb Tests were passing for me before I merged this several times, and it looks like the old logs are gone. Did the issue disappear after the commit was reverted? Does look suspicious I admit. This has been running without issue in our prod. Let me try to run this 10 or so times in CI and see if it's still failing, if it is not I propose remerging, and at least getting a traceback next time it fails in master CI to help fix (I can monitor CI for a while to make sure).

Also sorry for the late reply, need to figure out why I'm not getting email notifications anymore...

**Comment by ashb** [comment]
:D I don't remember exactly what was going on anymore, sorry. It was probable this PR wasn't at fault

**Comment by aoen** [comment]
@ashb Got it, I think it was a reasonable course of action, mind taking a look at the new PR? It's a lot tinier since non-RBAC has been deprecated, and the testing framework has been made simpler: https://github.com/apache/airflow/pull/5339

**Pull request merged**